src/lib/formatters.js

```diff
@@ -9,3 +9,67 @@ export function dateTimeFormat(timestamp, options) {
 
   return new Intl.DateTimeFormat(undefined, chosenOptions).format(dateTime);
 }
+
+export function serializeFilters(searchParams, updateFilter) {
+  Object.keys(updateFilter).forEach((key) => {
+    let searchParamVal = null;
+
+    if (
+      key === 'id' ||
+      key === 'viewType' ||
+      key === 'type' ||
+      key === 'name'
+    ) {
+      searchParamVal = JSON.stringify([...updateFilter[key].exclude]);
+    } else if (key === 'timeStamp') {
+      searchParamVal = JSON.stringify(updateFilter[key]);
+    } else {
+      searchParamVal = updateFilter[key];
+    }
+
+    searchParams.set(key, searchParamVal);
+  });
+
+  return searchParams;
+}
+
+function getJSONParseVal(str) {
+  try {
+    return JSON.parse(str);
+  } catch (error) {
+    return null;
+  }
+}
+
+export function deSerializeFilters(searchParams) {
+  const filterIds = searchParams.get('id');
+  const filterViewTypes = searchParams.get('viewType');
+  const filterTypes = searchParams.get('type');
+  const filterNames = searchParams.get('name');
+  const filterKeyword = searchParams.get('keyword');
+  const filterTimestamp = searchParams.get('timeStamp');
+  const filterTabId = parseInt(searchParams.get('tabId'), 10) || null;
+
+  const ids = getJSONParseVal(filterIds);
+  let viewTypes = getJSONParseVal(filterViewTypes);
+  // converting empty string to undefined
```

Comment:
This comment isn't accurate any more. Remove it.

Reply:
done :-)